Lib/pathlib.py

```diff
@@ -445,6 +445,8 @@ def symlink(a, b, target_is_directory):
 
     # Helper for resolve()
```

Comment:
I'd suggest implementing this like `lchmod` above, i.e.:

```python
if hasattr(os, "readlink"):
    readlink = os.readlink
else:
    def readlink(self, path):
        raise NotImplementedError("readlink() not available on this system")
```

This way you only do the `hasattr` check once.

Reply:
Right, it makes sense